### PR TITLE
Feature/#103 bulk get seriestypes

### PR DESCRIPTION
**Pull request by fredrik-exini**

### CODE REVIEW

File: src/main/scala/se/nimsa/sbx/app/JsonFormats.scala [hunk 1]
```diff
@@ -138,14 +140,16 @@ trait JsonFormats {
   implicit val queryOrderFormat: Format[QueryOrder] = Json.format[QueryOrder]
   implicit val queryPropertyFormat: Format[QueryProperty] = Json.format[QueryProperty]
   implicit val queryFiltersFormat: Format[QueryFilters] = Json.format[QueryFilters]
-  implicit val queryFormat: Format[Query] = Json.format[Query]
-
+  implicit val queryFormat: Format[MetaDataProtocol.Query] = Json.format[MetaDataProtocol.Query]
```
Comment (fredrik-exini) [inline src/main/scala/se/nimsa/sbx/app/JsonFormats.scala]: you are so right my good Sherlock

File: src/main/scala/se/nimsa/sbx/app/routing/SeriesTypeRoutes.scala [hunk 2]
```diff
@@ -138,7 +139,19 @@ trait SeriesTypeRoutes { this: SliceboxBase =>
             }
           }
         }
-
+      } ~ pathPrefix("series") {
+        path("query") {
+          pathEndOrSingleSlash {
```
Comment (KarlSjostrand) [inline src/main/scala/se/nimsa/sbx/app/routing/SeriesTypeRoutes.scala]: The `pathPrefix` then `path` then `pathEndOrSingleSlash` can be contracted to a single line of `path("series" / "query")`. `path` is terminal so there is no need for a `pathEndOrSingleSlash` after a `path`, only after a `pathPrefix`.

File: src/main/scala/se/nimsa/sbx/seriestype/SeriesTypeDAO.scala [hunk 3]
```diff
@@ -187,4 +187,14 @@ class SeriesTypeDAO(val dbConf: DatabaseConfig[JdbcProfile])(implicit ec: Execut
     } yield st
     innerJoin.result
   }
+
+  def seriesTypesForListOfSeries(seriesIds: Seq[Long]): Future[SeriesIdSeriesTypesResult] = db.run {
```
Comment (KarlSjostrand) [inline src/main/scala/se/nimsa/sbx/seriestype/SeriesTypeDAO.scala]: Sorry for nitpicking. Can this method return `Future[Seq[SeriesIdSeriesType]]` instead and leave the wrapping in a `SeriesIdSeriesTypesResult` to `SeriesTypeServiceActor`?

File: src/main/scala/se/nimsa/sbx/seriestype/SeriesTypeServiceActor.scala [hunk 4]
```diff
@@ -117,6 +117,9 @@ class SeriesTypeServiceActor(seriesTypeDao: SeriesTypeDAO)(implicit timeout: Tim
             val seriesTypes = getSeriesTypesForSeries(seriesId)
             sender ! SeriesTypes(seriesTypes)
 
+          case GetSeriesTypesForListOfSeries(seriesIds) =>
+            sender ! getSeriesTypesForListOfSeries(seriesIds)
```
Comment (KarlSjostrand) [inline src/main/scala/se/nimsa/sbx/seriestype/SeriesTypeServiceActor.scala]: This is where wrapping would happen...

**Pull request merged**